XdmfWriter.h

```diff
@@ -527,6 +532,51 @@ class XdmfWriter
 				<< "\" GridType=\"Uniform\">";
 	}
 
+	/**
+	 * Backup an existing xdmf file
+	 */
+	void backup(const std::string &prefix)
+	{
+		std::string fileName = prefix + ".xdmf";
+		// Backup any existing file
+		struct stat statBuffer;
+		if (m_rank == 0 && stat(fileName.c_str(), &statBuffer) == 0) {
+			logWarning() << fileName << "already exists. Creating backup.";
+			if (!m_backupTimeStamp.has_value()) {
+				m_backupTimeStamp = utils::TimeUtils::timeAsString("%Y-%m-%d_%H-%M-%S", time(0L));
```

Comment:
should you instead raise an error?
I mean this should never happen in practice, right?